src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/PooledObjects/PooledArrayBuilder`1.cs

```diff
@@ -68,6 +70,16 @@ public void ClearAndFree()
         }
     }
 
+    private readonly ImmutableArray<T>.Builder CreateBuilder()
+    {
+        var result = _pool.Get();
+        if (_capacity.HasValue)
+        {
+            result.SetCapacityIfNeeded(_capacity.Value);
+        }
```

Comment:
Use pattern matching to avoid the implicit double-check of `HasValue`.

```suggestion
        if (_capacity is int capacity)
        {
            result.SetCapacityIfNeeded(capacity);
        }
```


(Nullable<T>.Value has to test whether there's a value in order to throw if it doesn't. Note: `GetValueOrDefault()` is faster if you know there's a value because it just avoids the check and returns the underlying value field.)

Reply:
I checked this in sharplab at the time of writing, and saw the `GetValueOrDefault()`call and assumed it would still have to do the check. Thank you for the info, I should have dug deeper